util/dvsim/Launcher.py

```diff
@@ -274,19 +276,13 @@ def _get_n_lines(pos, num):
         # exit code for whatever reason, then show the last 10 lines of the log
         # as the failure message, which might help with the debug.
         if self.exit_code != 0:
-            err_msg = ("Job returned non-zero exit code:\nLast 10 lines:\n"
-                       "```\n{}\n```\n")
-            err_msg = err_msg.format(''.join(lines[-10:]).strip())
-            return "F", err_msg
-
-        # Ensure all pass patterns were seen.
+            return "F", ErrorMessage(line_number=len(lines)-10,
+                                     message="Job returned non-zero exit code")
```

Comment:
This is asking for trouble: the bucket is determined based on the ErrorMessage.message, and for infra failures it is best to provide the real diagnostic (like a non-zero exit code) instead of passing a somewhat random and very long multi-line string that may say nothing about what the real failure was, or that could split non-zero exit into multiple buckets.

I think if we want to pass additional lines for a failure it should be done in such a way that these lines don't determine the bucket, or we will end up loosing information. I suppose if we pass multiple lines we can just bucketize based on the first line.

Reply:
Okay, lets revisit this later.